### ISSUE TITLE
Script using ProactorEventLoop does not exit cleanly when subprocesses are spawned.

### ISSUE THREAD

**Issue by ab23bab0-f908-4939-805f-b62b168aa6c5** [python/cpython#81562]
BPO | [37381](https://bugs.python.org/issue37381)
--- | :---
Nosy | @pfmoore, @tjguk, @asvetlov, @zware, @1st1, @zooba, @cr1901

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = None
created_at = <Date 2019-06-23.22:35:46.548>
labels = ['type-bug', '3.7', 'OS-windows', 'expert-asyncio']
title = 'Script using ProactorEventLoop does not exit cleanly when subprocesses are spawned.'
updated_at = <Date 2019-06-24.01:07:24.313>
user = 'https://github.com/cr1901'
```

bugs.python.org fields:
```python
activity = <Date 2019-06-24.01:07:24.313>
actor = 'cr1901'
assignee = 'none'
closed = False
closed_date = None
closer = None
components = ['Windows', 'asyncio']
creation = <Date 2019-06-23.22:35:46.548>
creator = 'cr1901'
dependencies = []
files = []
hgrepos = []
issue_num = 37381
keywords = []
message_count = 2.0
messages = ['346345', '346346']
nosy_count = 7.0
nosy_names = ['paul.moore', 'tim.golden', 'asvetlov', 'zach.ware', 'yselivanov', 'steve.dower', 'cr1901']
pr_nums = []
priority = 'normal'
resolution = None
stage = None
status = 'open'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue37381'
versions = ['Python 3.5', 'Python 3.6', 'Python 3.7']
```

</p></details>

**Comment by arhadthedev** [comment]
This issue may be closed because the bug is fixed:

- gh-92841

**Comment by arhadthedev** [comment]
If this issue is fixed by gh-92841, it should be closed.

**Comment by kumaraditya303** [comment]
@cr1901 Can you check if this is still an issue on main branch and 3.11? It would be better if you can provide a reproducer without involving rust.

**Comment by kumaraditya303** [comment]
`asyncio` subprocess is significantly improved in 3.11 and 3.12 (main). A lot of bugs have been fixed so it is likely the bug is already fixed. I look forward to a Python only reproducer.

**Comment by kumaraditya303** [comment]
I tried your rust reproducer and the bug is fixed in main, 3.11 and 3.10 it doesn't hangs so closing as it is fixed.
